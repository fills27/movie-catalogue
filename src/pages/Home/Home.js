import React, {Component} from 'react'

class Home extends Component {
  render() {
    return (
      <div className='center-place'>
        <h1>
React Scratch by
          <a rel='noopener noreferrer' target='_blank' href='https://www.linkedin.com/in/syafilm/'>&nbsp;Syafil M</a>
        </h1>
      </div>
    )
  }
}

export default Home
