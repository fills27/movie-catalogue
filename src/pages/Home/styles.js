import {css} from '@emotion/css'

const topRated = css`
  color: #fff;
`

const sliderItem = css`
  display: inline-block !important;
  position: relative !important;
  img{
    width: 100%;
    margin: 0px;
    object-fit: cover;
    height: auto;
  }
`

const parentSliderItem = (label) => {

  const labelStyle = label !== 'nowPlaying' ?
  `padding-bottom: 130%;`
  :
  `padding-bottom: 72%;`

  return css`
    padding: 5px;
    border-radius: 2px;
    background: rgba(255, 255, 255, 0.6);
    > div{
      width: 100%;
      ${labelStyle}
      height: 0px;
      border-radius: 2px;
      overflow: hidden;
      position: relative;
    }
  `
}

const wrapperSlider = css`
  padding-right: 20px;
  text-decoration: none;
  h3{
    margin-top: 0px;
    color: #000;
    font-weight: bold;
    overflow: hidden !important;
    display: -webkit-box !important;
    -webkit-box-orient: vertical !important;
    -webkit-line-clamp: 1 !important;
  }
  span{
    color: rgba(0,0,0,0.5);
    font-weight: bold;
  }
`

const contentSlider = css`
  padding: 10px;
`

const hero = (url) => {
  return css`
    background-size: cover;
    background-repeat: no-repeat;
    background-image: url(${url});
    background-position: right -300px top;
    width: 100%;
  `
}

const overlayhero = css`
  padding-top: 100px;
  padding-bottom: 50px;
  background: linear-gradient(to right, rgba(2, 37, 64, 1.00) 300px, rgba(2, 37, 64, 0.64) 100%);
  height: 500px;
  h1{
    font-size: 30px;
    color: #fff;
    margin-bottom: 30px;
  }
  b{
    color: #fff;
    display: block;
    margin-bottom: 10px;
    font-size: 18px;
  }
  a{
    margin-top: 20px;
    color: #fff;
    font-size: 18px;
    display: inline-block;
  }
`

const rating = css`
  font-size: 22px;
  color: #fff;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
`

const overview = css`
  font-size: 18px;
  color: rgba(255,255,255, 0.4);
  display: block;
`

export default{
  sliderItem,
  parentSliderItem,
  topRated,
  wrapperSlider,
  contentSlider,
  hero,
  overlayhero,
  rating,
  overview
}