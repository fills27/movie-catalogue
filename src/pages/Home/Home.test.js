import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

import Home from './Home'

describe('Home page', () => {
  it('Can render Smoothly', () => {
    const component = shallow(
      <Home />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })
})
