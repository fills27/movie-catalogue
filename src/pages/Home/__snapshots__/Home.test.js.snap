// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Can render Smoothly 1`] = `
<div
  className="center-place"
>
  <h1>
    React Scratch by
    <a
      href="https://www.linkedin.com/in/syafilm/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Â Syafil M
    </a>
  </h1>
</div>
`;
